src/lib/pubkey/classic_mceliece/cmce.h

```diff
@@ -0,0 +1,145 @@
+ * It is endorsed by the German Federal Office for Information Security (BSI) for its conservative security
+ * assumptions and a corresponding draft for an ISO standard has been prepared. Both NIST and ISO parameter
+ * sets are implemented here. See https://classic.mceliece.org/ for the specifications and other details.
+ *
+ * Advantages of Classic McEliece:
+ * - Conservative post-quantum security assumptions
+ * - Very fast encapsulation
+ * - Fast decapsulation
+ *
+ * Disadvantages of Classic McEliece:
+ * - Very large public keys (0.26 MB - 1.36 MB)
+ * - Relatively slow key generation
+ * - Algorithm is complex and hard to implement side-channel resistant
+ */
+class BOTAN_PUBLIC_API(3, 4) Classic_McEliece_PublicKey : public virtual Public_Key {
+   public:
+      /**
+       * @brief Load a Classic McEliece public key from bytes.
+       *
+       * @param alg_id The algorithm identifier
+       * @param key_bits The public key bytes
+       */
+      Classic_McEliece_PublicKey(const AlgorithmIdentifier& alg_id, std::span<const uint8_t> key_bits);
+
+      /**
+       * @brief Load a Classic McEliece public key from bytes.
+       *
+       * @param key_bits The public key bytes
+       * @param param_set The parameter set
+       */
+      Classic_McEliece_PublicKey(std::span<const uint8_t> key_bits, Classic_McEliece_Parameter_Set param_set);
+
+      Classic_McEliece_PublicKey(const Classic_McEliece_PublicKey& other);
+      Classic_McEliece_PublicKey& operator=(const Classic_McEliece_PublicKey& other);
+      Classic_McEliece_PublicKey(Classic_McEliece_PublicKey&&) = default;
+      Classic_McEliece_PublicKey& operator=(Classic_McEliece_PublicKey&&) = default;
+
+      ~Classic_McEliece_PublicKey() override = default;
+
+      std::string algo_name() const override { return "ClassicMcEliece"; }
+
+      AlgorithmIdentifier algorithm_identifier() const override;
+
+      OID object_identifier() const override;
+
+      size_t key_length() const override;
+
+      size_t estimated_strength() const override;
+
+      std::vector<uint8_t> public_key_bits() const override;
+
+      std::vector<uint8_t> raw_public_key_bits() const override;
+
+      bool check_key(RandomNumberGenerator&, bool) const override;
+
+      bool supports_operation(PublicKeyOperation op) const override {
+         return (op == PublicKeyOperation::KeyEncapsulation);
+      }
+
+      std::unique_ptr<Private_Key> generate_another(RandomNumberGenerator& rng) const final;
+
+      std::unique_ptr<PK_Ops::KEM_Encryption> create_kem_encryption_op(std::string_view params,
+                                                                       std::string_view provider) const override;
+
+   protected:
+      Classic_McEliece_PublicKey() = default;
+
+   protected:
+      std::shared_ptr<Classic_McEliece_PublicKeyInternal>
+         m_public;  // NOLINT(misc-non-private-member-variables-in-classes)
```

Comment:
Readability nit I think this would be better with `// NOLINTNEXTLINE` so the declaration itself isn't split